### PR TITLE
Abstract Search Permission

### PR DESCRIPTION
**Pull request by infofromca**

### CODE REVIEW

**Comment by sebastienros** [comment]
Can you explain why you need this? I am afraid that if we start doing this someone is going to do that for every component, and I am not sure (yet) that we need this.

**Comment by infofromca** [comment]
@sebastienros 
If we need to create another new search which I think others need it, we can reuse this.

File: src/OrchardCore/OrchardCore.Lucene.Abstractions/SearchPermissionResult.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        public bool Succeeded { get; protected set; }
+
+        /// <summary>
+        /// Indicating a failed operation, with a list of errors if applicable.
+        /// </summary>
+        /// <param name="errors">An optional array of <see cref="LocalizedString"/> which caused the operation to fail.</param>
+        public static SearchPermissionResult Fail(params LocalizedString[] errors) => new SearchPermissionResult { Succeeded = false, Errors = errors, Failed = true, Forbided = false };
+
+        public bool Failed { get; protected set; }
+        public bool Forbided { get; protected set; }
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Lucene.Abstractions/SearchPermissionResult.cs:30]: Forbided => Forbidden ?

Comment (infofromca) [reply src/OrchardCore/OrchardCore.Lucene.Abstractions/SearchPermissionResult.cs]: @agriffard changed

**Comment by agriffard** [comment]
Error during build:
```
Controllers\SearchController.cs(72,17): error CS0103: The name 'permissions' does not exist in the current context [C:\projects\orchardcore\src\OrchardCore.Modules\OrchardCore.Lucene\OrchardCore.Lucene.csproj]
Controllers\SearchController.cs(74,71): error CS0103: The name 'permissions' does not exist in the current context [C:\projects\orchardcore\src\OrchardCore.Modules\OrchardCore.Lucene\OrchardCore.Lucene.csproj]
```

**Comment by Skrypt** [comment]
I don't dislike this PR but make it more generic then.

**Comment by agriffard** [comment]
Closing. Reopen if needed and conflicts are solved.

**Pull request closed without merging**